File: AndroidSdkDemo/app/src/main/java/com/exozet/demo/MainActivity.kt
package com.exozet.demo

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import io.mpos.accessories.AccessoryFamily
import io.mpos.accessories.parameters.AccessoryParameters
import io.mpos.provider.ProviderMode
import io.mpos.transactions.parameters.TransactionParameters
import io.mpos.ui.shared.MposUi
import io.mpos.ui.shared.model.MposUiConfiguration
import java.math.BigDecimal
import java.util.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // http://www.payworks.mpymnt.com/node/143
        paymentButtonClicked()
    }

    fun paymentButtonClicked() {
        val ui = MposUi.initialize(this, ProviderMode.MOCK,
                "merchantIdentifier", "merchantSecretKey")

        ui.configuration.summaryFeatures = EnumSet.of(
                // Add this line, if you do want to offer printed receipts
                // MposUiConfiguration.SummaryFeature.PRINT_RECEIPT,
                MposUiConfiguration.SummaryFeature.SEND_RECEIPT_VIA_EMAIL)

        // Start with a mocked card reader:
        val accessoryParameters = AccessoryParameters.Builder(AccessoryFamily.MOCK)
                .mocked()
                .build()
        ui.configuration.terminalParameters = accessoryParameters

        val transactionParameters = TransactionParameters.Builder()
                .charge(BigDecimal("5.00"), io.mpos.transactions.Currency.EUR)
                .subject("Bouquet of Flowers")
                .customIdentifier("yourReferenceForTheTransaction")
                .build()

        val intent = ui.createTransactionIntent(transactionParameters)
        startActivityForResult(intent, MposUi.REQUEST_CODE_PAYMENT)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {

        if (requestCode == MposUi.REQUEST_CODE_PAYMENT) {
            if (resultCode == MposUi.RESULT_CODE_APPROVED) {
                // Transaction was approved
                Toast.makeText(this, "Transaction approved", Toast.LENGTH_LONG).show()
            } else {
                // Card was declined, or transaction was aborted, or failed
                // (e.g. no internet or accessory not found)
                Toast.makeText(this, "Transaction was declined, aborted, or failed",
                        Toast.LENGTH_LONG).show()
            }
            // Grab the processed transaction in case you need it
            // (e.g. the transaction identifier for a refund).
            // Keep in mind that the returned transaction might be null
            // (e.g. if it could not be registered).
            val transaction = MposUi.getInitializedInstance().transaction
        }
    }
}

File: Demo2/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;
using IO.Mpos.UI.Shared;
using IO.Mpos.Provider;
using IO.Mpos.UI.Shared.Model;
using IO.Mpos.Accessories.Parameters;
using IO.Mpos.Accessories;
using IO.Mpos.Transactions.Parameters;
using Java.Math;
using Java.Util;
using Android.Content;
using Android.Util;

namespace Demo2
{
    [Activity(Label = "Demo2", MainLauncher = true, Icon = "@mipmap/icon")]
    public class MainActivity : Activity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            SetContentView(Resource.Layout.Main);

            Button button = FindViewById<Button>(Resource.Id.myButton);

            button.Click += delegate { Pay(); };
        }

        private void Pay()
        {
            MposUi ui = MposUi.Initialize(this, ProviderMode.Mock, "merchantIdentifier", "merchantSecretKey");
            ui.Configuration.SetSummaryFeatures(EnumSet.Of(
                // Add this line, if you do want to offer printed receipts
                // MposUiConfiguration.SummaryFeature.PRINT_RECEIPT,
                MposUiConfiguration.SummaryFeature.SendReceiptViaEmail)
            );

            //// Start with a mocked card reader:
            AccessoryParameters accessoryParameters = new AccessoryParameters.Builder(AccessoryFamily.Mock)
                .Mocked()
                .Build();
            ui.Configuration.SetTerminalParameters(accessoryParameters);

            var transactionParameters = new TransactionParametersBuilder()
                .Charge(new BigDecimal("5.00"), IO.Mpos.Transactions.Currency.Eur)
                .Subject("Bouquet of Flowers")
                .CustomIdentifier("yourReferenceForTheTransaction")
                .Build();

            Intent intent = ui.CreateTransactionIntent(transactionParameters);
            StartActivityForResult(intent, MposUi.RequestCodePayment);
        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            base.OnActivityResult(requestCode, resultCode, data);
            Log.Verbose("Demo", $"[OnActivityResult] {requestCode} {resultCode} {data}");
            if (requestCode == MposUi.RequestCodePayment)
                {
                //if (resultCode == MposUi.ResultCodeApproved)
                    //{
                        // Transaction was approved
                Toast.MakeText(this, $"Transaction {resultCode}", ToastLength.Long).Show();
                    //}
                    //else
                    //{
                    // Card was declined, or transaction was aborted, or failed
                    // (e.g. no internet or accessory not found)
                    //Toast.MakeText(this, "Transaction was declined, aborted, or failed", ToastLength.Long).Show();
                    //}
                    // Grab the processed transaction in case you need it
                    // (e.g. the transaction identifier for a refund).
                    // Keep in mind that the returned transaction might be null
                    // (e.g. if it could not be registered).
                     var transaction = MposUi.InitializedInstance.Transaction;
                }
        }

    }

}